ion-schema/src/lib.rs

```diff
@@ -359,3 +360,30 @@ impl UserReservedFields {
         Ok(())
     }
 }
+
+impl WriteToIsl for UserReservedFields {
+    fn write_to<W: IonWriter>(&self, writer: &mut W) -> IonSchemaResult<()> {
```

Comment:
_Very_ minor suggestion—there's a lot of stepping in and out in this function. It might be helpful to add comments or spacing to make it clear what bits belong together.